hbase_storage/src/main/java/com/splicemachine/access/client/MemstoreKeyValueScanner.java

```diff
@@ -59,16 +63,35 @@ public boolean advance(){
 
     public boolean nextResult() throws IOException{
         cellScannerIndex=0;
-        currentResult=this.resultScanner.next();
-        if(currentResult!=null){
-            cells=currentResult.rawCells();
-            peakKeyValue=(KeyValue)current();
-            rows++;
-            return true;
-        }else{
+        previousResult = currentResult;
+        currentResult = this.resultScanner.next();
+        if(currentResult == null) {
             // This shouldn't happen, throw exception and re-init the scanner
             throw new DoNotRetryIOException("Memstore scanner shouldn't end prematurely");
         }
+        if (currentResult.rawCells() != null && CellUtil.matchingFamily(currentResult.rawCells()[0], FLUSH) ) {
+            if (LOG.isTraceEnabled())
+                LOG.trace("CURRENT CELL: " + currentResult.rawCells()[0]);
+            assert currentResult.rawCells().length == 1;
+            if (previousResult != null && previousResult.rawCells() != null) {
+                assert previousResult.rawCells().length > 0;
+                KeyValue previousCell = (KeyValue) previousResult.rawCells()[previousResult.rawCells().length - 1];
+                if (LOG.isTraceEnabled())
+                    LOG.trace("PREVIOUS CELL: " + previousCell);
+                if (CellUtil.matchingRows(currentResult.rawCells()[0], HConstants.EMPTY_START_ROW)) {
+                    byte[] rowId = Bytes.add(
+                            CellUtil.cloneRow(previousCell),
+                            CellUtil.cloneRow(currentResult.rawCells()[0]));
```

Comment:
We should append a NULL byte, not the RowId of the Flush cell.

Reply:
Done